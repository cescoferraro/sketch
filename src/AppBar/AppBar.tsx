import * as React from "react";
import {
  AppBar as MUIAppBar,
  Box,
  IconButton,
  Toolbar,
  Typography,
} from "@mui/material";
import { SketchLogo } from "../assets/sketch-logo";
import { useNavigate } from "react-router-dom";

interface Props {
  title: string;
  children?: React.ReactNode;
}

export const AppBar = ({ title, children }: Props): React.ReactElement => {
  const navigate = useNavigate();
  return (
    <Box>
      <MUIAppBar
        position="static"
        sx={{
          background: "white",
          boxShadow: "none",
          borderBottom: "1px rgb(230,230,230) solid",
        }}
      >
        <Toolbar>
          {!children && (
            <span style={{ display: "flex", position: "absolute" }}>
              <IconButton
                size="large"
                edge="start"
                color="inherit"
                aria-label="menu"
                sx={{ mr: 2 }}
                onClick={() => navigate("/")}
              >
                <SketchLogo />
              </IconButton>
            </span>
          )}
          {children}
          <Typography
            variant="h6"
            component="div"
            align={"center"}
            sx={{ color: "black", flexGrow: 1 }}
          >
            {title}
          </Typography>
        </Toolbar>
      </MUIAppBar>
    </Box>
  );
};
