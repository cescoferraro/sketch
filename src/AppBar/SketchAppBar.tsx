import * as React from "react";
import { AppBar, Box, IconButton, Toolbar, Typography } from "@mui/material";
import { SketchLogo } from "../assets/sketch-logo";
import { useNavigate } from "react-router-dom";

interface Props {
  title: string;
  children?: React.ReactNode;
}

export const SketchAppBar = ({
  title,
  children,
}: Props): React.ReactElement => {
  const navigate = useNavigate();
  return (
    <Box>
      <AppBar
        position="static"
        sx={{
          background: "white",
          boxShadow: "none",
          borderBottom: "1px rgb(230,230,230) solid",
        }}
      >
        <Toolbar>
          {!children && (
            <IconButton
              size="large"
              edge="start"
              color="inherit"
              aria-label="menu"
              sx={{ mr: 2 }}
              onClick={() => navigate("/")}
            >
              <SketchLogo />
            </IconButton>
          )}
          {children}
          <Typography
            variant="h6"
            component="div"
            align={"center"}
            sx={{ color: "black", flexGrow: 1 }}
          >
            {title}
          </Typography>
        </Toolbar>
      </AppBar>
    </Box>
  );
};
