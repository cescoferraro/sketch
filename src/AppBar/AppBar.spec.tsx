import { AppBar } from "./AppBar";
import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

test("AppBar renders", () => {
  const tree = render(
    <BrowserRouter>
      <AppBar title={"Testing"} />
    </BrowserRouter>
  );
  expect(tree).toMatchSnapshot();
});
