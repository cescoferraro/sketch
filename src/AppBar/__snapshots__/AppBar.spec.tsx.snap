// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-gsDKAQ MuiBox-root"
      >
        <header
          class="MuiPaper-root-jrhEqT iplgfn MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root-gCtUal fMpyq MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic"
        >
          <div
            class="MuiToolbar-root-hKYalw hhGIQh MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular"
          >
            <span
              style="display: flex; position: absolute;"
            >
              <button
                aria-label="menu"
                class="MuiButtonBase-root-KjFju gUleGd MuiButtonBase-root MuiIconButton-root-jYvkJQ iZXrrt MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeLarge"
                tabindex="0"
                type="button"
              >
                <svg
                  height="25"
                  viewBox="0 0 27 25"
                  width="27"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <defs>
                    <path
                      d="M6.12869318,2.46533203 L13.5,1.6875 L20.8713068,2.46533203 C20.8713068,2.46533203 26.578125,10.1118164 26.578125,10 C26.578125,10.1118164 13.5,25.3125 13.5,25.3125 L0.421875,10 L6.12869318,2.46533203 Z"
                      id="sketch-logo-a"
                    />
                  </defs>
                  <g
                    fill="none"
                    fill-rule="evenodd"
                    transform="translate(0 -1)"
                  >
                    <mask
                      fill="#fff"
                      id="sketch-logo-b"
                    >
                      <use
                        href="#sketch-logo-a"
                      />
                    </mask>
                    <use
                      fill="#FDB300"
                      href="#sketch-logo-a"
                    />
                    <g
                      mask="url(#sketch-logo-b)"
                    >
                      <g
                        transform="translate(.422 9.112)"
                      >
                        <polygon
                          fill="#EA6C00"
                          points="5.297 .888 13.078 16.201 0 .888"
                        />
                        <polygon
                          fill="#EA6C00"
                          points="18.375 .888 26.156 16.201 13.078 .888"
                          transform="matrix(-1 0 0 1 39.234 0)"
                        />
                        <polygon
                          fill="#FDAD00"
                          points="5.297 .888 20.859 .888 13.078 16.201"
                        />
                      </g>
                    </g>
                    <g
                      mask="url(#sketch-logo-b)"
                    >
                      <g
                        transform="translate(.422 1.688)"
                      >
                        <polygon
                          fill="#FDD231"
                          points="13.078 0 5.707 .778 5.297 8.313"
                        />
                        <polygon
                          fill="#FDD231"
                          points="20.859 0 13.488 .778 13.078 8.313"
                          transform="matrix(-1 0 0 1 33.937 0)"
                        />
                        <polygon
                          fill="#FDAD00"
                          points="20.449 8.313 26.156 .778 25.748 8.313"
                          transform="matrix(-1 0 0 1 46.606 0)"
                        />
                        <polygon
                          fill="#FDAD00"
                          points="0 8.313 5.707 .778 5.297 8.313"
                        />
                        <polygon
                          fill="#FEEEB7"
                          points="13.078 0 5.297 8.313 20.857 8.313"
                        />
                      </g>
                    </g>
                  </g>
                </svg>
                <span
                  class="MuiTouchRipple-root-ieAhOC kYGcPf MuiTouchRipple-root"
                />
              </button>
            </span>
            <div
              class="MuiTypography-root-bWKFam gjPcUu MuiTypography-root MuiTypography-h6 MuiTypography-alignCenter"
            >
              Testing
            </div>
          </div>
        </header>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-gsDKAQ MuiBox-root"
    >
      <header
        class="MuiPaper-root-jrhEqT iplgfn MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root-gCtUal fMpyq MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic"
      >
        <div
          class="MuiToolbar-root-hKYalw hhGIQh MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular"
        >
          <span
            style="display: flex; position: absolute;"
          >
            <button
              aria-label="menu"
              class="MuiButtonBase-root-KjFju gUleGd MuiButtonBase-root MuiIconButton-root-jYvkJQ iZXrrt MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeLarge"
              tabindex="0"
              type="button"
            >
              <svg
                height="25"
                viewBox="0 0 27 25"
                width="27"
                xmlns="http://www.w3.org/2000/svg"
              >
                <defs>
                  <path
                    d="M6.12869318,2.46533203 L13.5,1.6875 L20.8713068,2.46533203 C20.8713068,2.46533203 26.578125,10.1118164 26.578125,10 C26.578125,10.1118164 13.5,25.3125 13.5,25.3125 L0.421875,10 L6.12869318,2.46533203 Z"
                    id="sketch-logo-a"
                  />
                </defs>
                <g
                  fill="none"
                  fill-rule="evenodd"
                  transform="translate(0 -1)"
                >
                  <mask
                    fill="#fff"
                    id="sketch-logo-b"
                  >
                    <use
                      href="#sketch-logo-a"
                    />
                  </mask>
                  <use
                    fill="#FDB300"
                    href="#sketch-logo-a"
                  />
                  <g
                    mask="url(#sketch-logo-b)"
                  >
                    <g
                      transform="translate(.422 9.112)"
                    >
                      <polygon
                        fill="#EA6C00"
                        points="5.297 .888 13.078 16.201 0 .888"
                      />
                      <polygon
                        fill="#EA6C00"
                        points="18.375 .888 26.156 16.201 13.078 .888"
                        transform="matrix(-1 0 0 1 39.234 0)"
                      />
                      <polygon
                        fill="#FDAD00"
                        points="5.297 .888 20.859 .888 13.078 16.201"
                      />
                    </g>
                  </g>
                  <g
                    mask="url(#sketch-logo-b)"
                  >
                    <g
                      transform="translate(.422 1.688)"
                    >
                      <polygon
                        fill="#FDD231"
                        points="13.078 0 5.707 .778 5.297 8.313"
                      />
                      <polygon
                        fill="#FDD231"
                        points="20.859 0 13.488 .778 13.078 8.313"
                        transform="matrix(-1 0 0 1 33.937 0)"
                      />
                      <polygon
                        fill="#FDAD00"
                        points="20.449 8.313 26.156 .778 25.748 8.313"
                        transform="matrix(-1 0 0 1 46.606 0)"
                      />
                      <polygon
                        fill="#FDAD00"
                        points="0 8.313 5.707 .778 5.297 8.313"
                      />
                      <polygon
                        fill="#FEEEB7"
                        points="13.078 0 5.297 8.313 20.857 8.313"
                      />
                    </g>
                  </g>
                </g>
              </svg>
              <span
                class="MuiTouchRipple-root-ieAhOC kYGcPf MuiTouchRipple-root"
              />
            </button>
          </span>
          <div
            class="MuiTypography-root-bWKFam gjPcUu MuiTypography-root MuiTypography-h6 MuiTypography-alignCenter"
          >
            Testing
          </div>
        </div>
      </header>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
